plugins/modules/ec2_instance.py

```diff
@@ -1703,7 +1704,8 @@ def pretty_instance(i):
 
 
 def determine_iam_role(name_or_arn):
-    if re.match(r'^arn:aws:iam::\d+:instance-profile/[\w+=/,.@-]+$', name_or_arn):
+    result = parse_aws_arn(name_or_arn)
+    if result and result['service'] == 'iam' and re.match(r"instance-profile/[\w+=/,.@-]+$", result['resource']):
```

Comment:
```suggestion
    if result and result['service'] == 'iam' and result['resource'].startswith('instance-profile/'):
```

How about simplifying it a little rather than using a full blown regex?

Reply:
I tried to keep the same behavior I don't know if the regex was intended, but I totally agree with you this can be simplified